Google.PowerShell.IntegrationTests/BigQuery/BigQuery.BqDataset.Tests.ps1

```diff
@@ -120,16 +143,64 @@ Describe "Set-BqDataset" {
     } 
 
     It "should add when a dataset does not exist" {
-        $data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
-        $data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
-        $data.DatasetReference.DatasetId = "test_dataset_id7"
-        $data.DatasetReference.ProjectId = $project
-        $set = Set-BqDataset $data
-        $set.DatasetReference.DatasetId | Should Be "test_dataset_id7"
+        try {
+            $data = New-Object -TypeName Google.Apis.Bigquery.v2.Data.Dataset
+            $data.DatasetReference = New-Object -TypeName Google.Apis.Bigquery.v2.Data.DatasetReference
+            $data.DatasetReference.DatasetId = "test_dataset_id7"
+            $data.DatasetReference.ProjectId = $project
+            $set = Set-BqDataset $data
+            $set.DatasetReference.DatasetId | Should Be "test_dataset_id7"
+        } finally {
+            Get-BqDataset "test_dataset_id7" | Remove-BqDataset
+        }
     } 
+    
+    It "should label things properly" {
+        try {
+            $data = New-BqDataset "test_label_1"
+            $data = $data | Set-BqDataset -SetLabel "test" "one"
+            $data.Labels.Count | Should Be 1
+            $data.Labels["test"] | Should Be "one"
+        } finally {
+            Get-BqDataset "test_label_1" | Remove-BqDataset
+        }
+    }
+
+    It "should handle a lot of labels at once" {
+        try {
+            $data = New-BqDataset "test_label_2"
+            $data = $data | Set-BqDataset -SetLabel "test","other","third" "one","two","three"
+            $data.Labels.Count | Should Be 3
+            $data.Labels["test"] | Should Be "one"
+            $data.Labels["other"] | Should Be "two"
+            $data.Labels["third"] | Should Be "three"
+        } finally {
+            Get-BqDataset "test_label_2" | Remove-BqDataset
+        }
+    }
 
-    #TODO(ahandley): Find reason behind occasional (25%) 412 Precondition error (If-Match - header).
+    It "should clear just one label" {
+        try {
+            $data = New-BqDataset "test_label_3"
+            $data = $data | Set-BqDataset -SetLabel "test","other" "one","two"
+            $data = $data | Set-BqDataset -ClearLabel "test"
+            $data.Labels.Count | Should Be 1
+            $data.Labels["other"] | Should Be "two"
+        } finally {
+            Get-BqDataset "test_label_3" | Remove-BqDataset
+        }
+    }
 
+    It "should clear multiple labels" {
+        try {
+            $data = New-BqDataset "test_label_4"
+            $data = $data | Set-BqDataset -SetLabel "test","other" "one","two"
+            $data = $data | Set-BqDataset -ClearLabel "test", "other"
+            $data.Labels | Should Be $null
+        } finally {
+            Get-BqDataset "test_label_4" | Remove-BqDataset
+        }
+    }
 }
```

Comment:
I tested that in-person.  Should have put that in as well, but yeah.  Went around the issue.